backend/users/views.py

```diff
@@ -243,3 +243,38 @@ def change_password_view(request):
         return render(request, "users/change_password.html", {"error": error})
 
     return render(request, "users/change_password.html")
+
+
+##############
+##### Frineds Methods
+##############
+
+
+@jwt_login_required
+def add_friend(request, user_id):
+    friend = get_object_or_404(CustomUser, id=user_id)
+    request.user.add_friend(friend)
+    return redirect("user_profile", user_id=user_id)
+
+
+@jwt_login_required
+def remove_friend(request, user_id):
+    friend = get_object_or_404(CustomUser, id=user_id)
+    request.user.remove_friend(friend)
+    return redirect("user_profile", user_id=user_id)
+
+
+@jwt_login_required
+def friend_list(request):
+    friends = request.user.get_friends()
+    return render(request, "users/friend_list.html", {"friends": friends})
```

Comment:
this html file is not in this PR

Reply:
exactly, it depends on frontend, i assumed these two pages will be created by you

Reply:
alright that makes sense